src/glang/codegen/builtin_types.py

```diff
@@ -329,6 +329,28 @@ def __init__(self) -> None:
         super().__init__("iptr", get_ptr_type_info().size.in_bits, True)
 
 
+class IEEEFloatDefinition(PrimitiveDefinition):
+    # NOTE: we use IEEE rather than the target specific SIMD types
```

Comment:
Does the front end need to be target aware then when compiling floating point code?

Reply:
Hmm, probably yes... Good point.... I think I need to add something to our `targets/*.yml`

Reply:
That would do, although we could just hold off until we actually need to introduce a distinct implementation (a quick search indicates the IEEE representation is very widely used)